actions/src/main/java/com/redhat/service/bridge/actions/webhook/WebhookAction.java

```diff
@@ -0,0 +1,49 @@
+package com.redhat.service.bridge.actions.webhook;
+
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.redhat.service.bridge.actions.ActionInvoker;
+import com.redhat.service.bridge.actions.ActionProvider;
+import com.redhat.service.bridge.actions.ActionProviderException;
+import com.redhat.service.bridge.infra.models.actions.BaseAction;
+import com.redhat.service.bridge.infra.models.dto.ProcessorDTO;
+
+import okhttp3.OkHttpClient;
+
+@ApplicationScoped
+public class WebhookAction implements ActionProvider {
+
+    public static final String TYPE = "Webhook";
+    public static final String ENDPOINT_PARAM = "endpoint";
+
+    @Inject
+    WebhookActionValidator validator;
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    public WebhookActionValidator getParameterValidator() {
+        return validator;
+    }
+
+    @Override
+    public ActionInvoker getActionInvoker(ProcessorDTO processor, BaseAction baseAction) {
+        String endpoint = Optional.ofNullable(baseAction.getParameters().get(ENDPOINT_PARAM))
+                .orElseThrow(() -> buildNoEndpointException(processor));
+        // customize http client here if needed (e.g. timeouts)
+        OkHttpClient client = new OkHttpClient.Builder().build();
+        return new WebhookInvoker(endpoint, client);
```

Comment:
I think this can be `@Inject` in the same way you do for the `WebhookActionValidator` which means we don't need to create an instance per-invocation of the Action.

Reply:
Same comment as [here](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/128#discussion_r750984913)

Reply:
OK thanks @kostola . Your comment on keeping things consistent is a good one. Let's go with it as it is 👍